src/models/db/accounts.js

```diff
@@ -0,0 +1,42 @@
+'use strict'
+
+const Account = require('./account').Account
+const db = require('../../services/db')
+
+function * getAccounts () {
+  return (yield Account.findAll())
+}
+
+function * getConnectorAccounts () {
+  return (yield Account.findAll({
+    where: { connector: { $ne: null } }
+  }))
+}
+
+function * getAccount (name) {
+  return (yield Account.findByName(name))
+}
+
+function * upsertAccount (account) {
+  // SQLite's implementation of upsert does not tell you whether it created the
+  // row or whether it already existed. Since we need to know to return the
+  // correct HTTP status code we unfortunately have to do this in two steps.
+  let existed
+  yield db.transaction(function * (transaction) {
+    existed = yield Account.findByName(account.name, { transaction })
+    if (existed) {
+      existed.setDataExternal(account)
+      yield existed.save({ transaction })
+    } else {
+      yield Account.createExternal(account, { transaction })
+    }
+  })
+  return existed
```

Comment:
I think it makes sense to return a boolean here for consistency with sequelize's `upsert` method.